lib/semian/circuit_breaker.rb

```diff
@@ -1,13 +1,35 @@
 module Semian
-  class CircuitBreaker
-    attr_reader :state
-
-    def initialize(exceptions:, success_threshold:, error_threshold:, error_timeout:)
+  class CircuitBreaker #:nodoc:
+    def initialize(name, exceptions:, success_threshold:, error_threshold:, error_timeout:, permissions: 0660)
+      @name = name.to_s
```

Comment:
It's so that the individual structures have a name. If you want to invert it, my argument against that would be that you'd need to have this code:
```
::Semian::AtomicEnum.new(
        name,
        permissions,
        [:closed, :half_open, :open])
```
outside the circuit breaker class and would be that wouldn't be great.